### PR TITLE
OCPBUGS-8449: Azure: don't set default subscriptionID for disk encryption sets

### PR DESCRIPTION
**Pull request by jhixson74**
https://issues.redhat.com/browse/OCPBUGS-8449

### CODE REVIEW

**Review by r4f4 — APPROVED** [review]
Pre-merge testing passed.
/approve
/lgtm

File: pkg/types/azure/validation/disk.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ func ValidateDiskEncryption(p *azure.MachinePool, cloudName azure.CloudEnvironme
 	if diskEncryptionSet != nil && cloudName == azure.StackCloud {
 		return append(allErrs, field.Invalid(childFldPath.Child("diskEncryptionSet"), diskEncryptionSet, "disk encryption sets are not supported on this platform"))
 	}
-	if diskEncryptionSet.SubscriptionID != "" && !RxSubscriptionID.MatchString(diskEncryptionSet.SubscriptionID) {
+	if !RxSubscriptionID.MatchString(diskEncryptionSet.SubscriptionID) {
```
Comment (patrickdillon) [inline pkg/types/azure/validation/disk.go:34]: nit: `field.Required` would be a more appropriate error when `diskEncryptionSet.SubscriptionID == ""`.

Comment (jhixson74) [reply pkg/types/azure/validation/disk.go:37]: Fixed.

**Pull request merged**